tests/expecter.go

```diff
@@ -228,7 +229,14 @@ func ExpectBatchWithValidatedSend(expecter expect.Expecter, batch []expect.Batch
 
 			// Remove the \n since it is translated by the console to \r\n.
 			previousSend = strings.TrimSuffix(previousSend, "\n")
-			bExp.R = fmt.Sprintf("%s%s%s", previousSend, "((?s).*)", bExp.R)
+			// Somtimes after logging in, the previous command will be removed from the buffer,
+			// for that we allow it also to match without the previous command, but only if it is at
+			// the beggining of the line.
+			// Plus in some consoles the prompt is read with a preceded white space, and sometimes it doesn't, so we
+			// allow that as well.
+			bExp.R = fmt.Sprintf("(%s%s%s"+`|^\s?`+"%s*%s)", previousSend, "((?s).*)", bExp.R, CRLF, bExp.R)
```

Comment:
I'm no regex expert, but is there a way to maybe combine the two options to 1 regex expression?

Reply:
Me neither :) 
Perhaps @jean-edouard  has a better Idea, but that's the best I could think of. What I was tring here is to  attach the previous BSend *o*r find the desired expression only at the beginning of the line\string (when the buffer is flushed before the expected command is getting in the buffer)